.github/workflows/cloud-data-sources-code-coverage.yml

```diff
@@ -0,0 +1,81 @@
+name: Cloud data sources test code coverage
+# Please be aware that this report targets cloud data source paths -- may not be as accurate as full coverage reports
+on:
+  pull_request:
+    paths:
+      - 'pkg/tsdb/azuremonitor/**'
+      - 'pkg/tsdb/cloudwatch/**'
+      - 'pkg/tsdb/cloudmonitoring/**'
+      - 'public/app/plugins/datasource/grafana-azure-monitor-datasource/**'
+      - 'public/app/plugins/datasource/cloudwatch/**'
+      - 'public/app/plugins/datasource/cloud-monitoring/**'
+    branches-ignore:
+      - dependabot/**
+      - backport-**
+
+jobs:
+  comment-coverage-difference:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-node@v2
+        with:
+          node-version: 17.x
+          cache: yarn
+      - name: yarn install for caller's PR
+        run: yarn install
+      - uses: actions/setup-go@v2
+        with:
+          go-version: '1.17.8'
+      - name: generate go for github.com/grafana/grafana
+        run: make gen-go
+
+      - name: Get Go coverage for PR
+        run: |
+          go test -v -coverpkg=./... -coverprofile=profile.cov ./pkg/tsdb/azuremonitor/... ./pkg/tsdb/cloudmonitoring/... ./pkg/tsdb/cloudwatch/...
+          echo "go-pr-cov=$(go tool cover -func profile.cov | grep total: | awk '{print substr($NF,1,length($NF)-1)}')" >> $GITHUB_ENV
+      - name: Get TypeScript coverage for PR
+        run: |
+          echo "ts-pr-cov=$(yarn test:coverage -- ./public/app/plugins/datasource/cloudwatch/** ./public/app/plugins/datasource/cloud-monitoring/** ./public/app/plugins/datasource/grafana-azure-monitor-datasource/** | grep 'All files' | awk '{print $10}')" >> $GITHUB_ENV
+      - uses: actions/checkout@v3
+        with:
+          ref: main
+      - name: yarn install for caller's main
+        run: yarn install
+      - name: generate go for github.com/grafana/grafana
+        run: make gen-go
+
+      - name: Get Go coverage on main
+        run: |
+          go test -v -coverpkg=./... -coverprofile=profile.cov ./pkg/tsdb/azuremonitor/... ./pkg/tsdb/cloudmonitoring/... ./pkg/tsdb/cloudwatch/...
+          echo "go-main-cov=$(go tool cover -func profile.cov | grep total: | awk '{print substr($NF,1,length($NF)-1)}')" >> $GITHUB_ENV
+      - name: Get TypeScript coverage on main
+        run: |
+          echo "ts-main-cov=$(yarn test:coverage -- ./public/app/plugins/datasource/cloudwatch/** ./public/app/plugins/datasource/cloud-monitoring/** ./public/app/plugins/datasource/grafana-azure-monitor-datasource/** | grep 'All files' | awk '{print $10}')" >> $GITHUB_ENV
```

Comment:
This is going way beyond what you need probably but I'll share just in case because I just learnt about it and find it super cool:

1. Add the following to `package.json` (`$*` acts as all params, somehow $1 did not work for me 🤷‍♀️ )
```
"test:coverage:datasource": "jest --coverage $* --collectCoverageFrom=\"$*/**/**/*.{ts,tsx}\"",
```
2. Use (here and line 39)
```
echo "ts-main-cov-cw=$(yarn test:coverage:datasource public/app/plugins/datasource/cloudwatch | grep 'All files' | awk '{print $10}')" >> $GITHUB_ENV
echo "ts-main-cov-az=$(yarn test:coverage:datasource public/app/plugins/datasource/grafana-azure-monitor-datasource | grep 'All files' | awk '{print $10}')" >> $GITHUB_ENV
echo "ts-main-cov-cm=$(yarn test:coverage:datasource public/app/plugins/datasource/cloud-monitoring | grep 'All files' | awk '{print $10}')" >> $GITHUB_ENV
```
3. You can now have 3 typescript columns for each of our plugin in the comment body 😃 

Bonus points if you can reuse the workflow but setting the datasource based on the `paths` so we'd only get the test coverage we care about 🤩 . (Really not needed but would be so cool 😉 )

Reply:
Thanks! I didn't realize the command I had was inaccurate.

For your command is it `yarn test public/app/plugins/datasource/grafana-azure-monitor-datasource public/app/plugins/datasource/cloudwatch public/app/plugins/datasource/cloud-monitoring --coverage --collectCoverageFrom="public/app/plugins/datasource/{grafana-azure-monitor-datasource,cloudwatch,cloud-monitoring}/**/**/*.{ts,tsx}"` ? (would that be the same as two trailing `**` by the way?)

I'm seeing that run I think all tests locally, not just for our three plugins. Also it ends with "Watch Usage" and a selection of choices. I don't think I ran it right.

Reply:
>  (would that be the same as two trailing ** by the way?)

You don't need double `**`, that already means any subfolder (with any depth)

Reply:
Nothing is needed at all? Was that `/**/**/*.{ts,tsx}` suffix needed then?

Reply:
Is it possible to get the sum of the three plugins' coverages in one command?

Reply:
no, I mean that `/**/**/` is the same than `/**/`, both means "any subfolder"

Reply:
> Is it possible to get the sum of the three plugins' coverages in one command?


That would be:
```
jest public/app/plugins/datasource/grafana-azure-monitor-datasource public/app/plugins/datasource/cloudwatch public/app/plugins/datasource/cloud-monitoring --coverage --collectCoverageFrom="public/app/plugins/datasource/{grafana-azure-monitor-datasource,cloudwatch,cloud-monitoring}/**/*.{ts,tsx}"

```
You would need to add it to `package.json` I believe.